lib/utils.pm

```diff
@@ -107,6 +108,10 @@ sub wait_boot {
             die "needle 'grub2' not found";
         }
     }
+    else {
+        # With Xen PV we don't see the graphical bootloader
```

Comment:
```
elsif (!(check_var('VIRSH_VMM_FAMILY', 'xen') && check_var('VIRSH_VMM_TYPE', 'linux'))) {
```
Maybe you missed the `!`? Or maybe I don't understand what you meant. I read the code as everything but `xen linux`. Then the last `else { ... }` is just `xen linux`.